Recon/include/Recon/Skims/EcalPreselectionSkimmer.h

```diff
@@ -0,0 +1,66 @@
+/**
+ * @file EcalPreselectionSkimmer.h
+ * @brief Processor used to pre-select events for the ECAL studies
+ * @author Tamas Almos Vami (UCSB)
+ */
+
+#ifndef RECON_SKIMS_EcalPreselectionSkimmer_H_
+#define RECON_SKIMS_EcalPreselectionSkimmer_H_
+
+//----------//
+//   LDMX   //
+//----------//
+#include "Ecal/Event/EcalVetoResult.h"
+#include "Framework/EventProcessor.h"
+#include "Tools/AnalysisUtils.h"
+
+namespace recon {
+
+class EcalPreselectionSkimmer : public framework::Producer {
+ public:
+  // @param parameters Set of parameters used to configure this processor.
+  virtual void configure(framework::config::Parameters &);
+
+  /** Constructor */
+  EcalPreselectionSkimmer(const std::string &name, framework::Process &process);
+
+  /** Destructor */
+  ~EcalPreselectionSkimmer() = default;
+
+  /**
+   * Run the processor and select events that pass pre-selection in ECAL
+   *
+   * @param event The event to process.
+   */
+  void produce(framework::Event &event);
```

Comment:
Just some restructuring (I like putting the functions in order that they are called except the constructor/desturcor) and adding `final`.

```suggestion
  /** Constructor */
  EcalPreselectionSkimmer(const std::string &name, framework::Process &process);

  /** Destructor */
  ~EcalPreselectionSkimmer() = default;

   // @param parameters Set of parameters used to configure this processor.
  virtual void configure(framework::config::Parameters &) final;

  /**
   * Run the processor and select events that pass pre-selection in ECAL
   *
   * @param event The event to process.
   */
  virtual void produce(framework::Event &event) final;
```

Reply:
done in next commit